### PR TITLE
Fixes #19205: Add proxy and port change support to rudder agent

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/19205

### CODE REVIEW

File: bin/rudder-client [hunk 1]
```diff
@@ -10,6 +10,8 @@ else
   CURL="curl"
 fi
 
+CONFFILE="/opt/rudder/etc/rudder.conf"
```
Comment (amousset) [inline bin/rudder-client:13]: agent.conf maybe?

**Comment by peckpeck** [comment]
PR updated with a new commit

**Comment by peckpeck** [comment]
ping @amousset

**Review by amousset — APPROVED** [review]

**Comment by peckpeck** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
needs a rebase

**Comment by peckpeck** [comment]
PR rebased

File: bin/rudder-client [hunk 2]
```diff
@@ -54,10 +54,15 @@ fi
 
 ## Get configuration
 PROXY=$(agent_conf https_proxy)
-HTTPS_PORT=$(agent_conf https_port)
 
-# get server from policy_server.dat
-SERVER=$(cut -d: -f1 "${RUDDER_VAR}/cfengine-community/policy_server.dat")
+# get server form configuration
+SERVER=$(get_agent_conf server)
```
Comment (amousset) [inline bin/rudder-client:59]:
```suggestion
SERVER=$(agent_conf server)
```

File: share/lib/common.sh [hunk 3]
```diff
@@ -193,6 +193,20 @@ agent_conf() {
   fi
 }
 
+# get port used to talk https with the server
+get_https_port() {
+  # get port from configuration
+  PORT=$(agent_conf https_port)
+  # if not trust the server on this
+  if [ "${PORT}" = "" ]; then
+    PORT=$(rudder_json_value 'SERVER_KEY_HASHES')
```
Comment (amousset) [inline share/lib/common.sh:202]:
```suggestion
    PORT=$(rudder_json_value 'POLICY_SERVER_KEY_HASH')
```

Comment (amousset) [inline share/lib/common.sh:202]: HTTPS_POLICY_DISTRIBUTION_PORT

**Comment by peckpeck** [comment]
PR updated with a new commit

**Review by amousset — APPROVED** [review]

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request merged**